### PR TITLE
[RV64_DYNAREC] Added more 0F opcodes for vector and optimized some opcodes too

### PR DESCRIPTION
**Pull request by ksco**

### CODE REVIEW

File: src/dynarec/rv64/dynarec_rv64_660f_vector.c [hunk 1]
```diff
@@ -149,14 +149,13 @@ uintptr_t dynarec64_660F_vector(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t i
...
             d1 = fpu_get_scratch(dyn);
-            VMAX_VX(d0, xZR, q0, VECTOR_UNMASKED);
-            VMAX_VX(d1, xZR, q1, VECTOR_UNMASKED);
             if (rv64_vlen >= 256) {
-                /*           mu            tu              sew           lmul=1 */
-                vtypei = (0b0 << 7) | (0b0 << 6) | (VECTOR_SEW16 << 3) | 0b000;
-                ADDI(x1, xZR, 16); // double the vl for slideup.
-                VSETVLI(xZR, x1, vtypei);
-                VSLIDEUP_VI(d0, 8, d1, VECTOR_UNMASKED); // splice d0 and d1 here!
+                vector_vsetvl_emul1(dyn, ninst, x1, VECTOR_SEW16, 2); // double the vl for slideup.
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_660f_vector.c:153]: there is no need to "un-double" v1?

File: src/dynarec/rv64/dynarec_rv64_00_0.c [hunk 2]
```diff
@@ -177,7 +178,11 @@ uintptr_t dynarec64_00_0(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
         case 0x0F:
             switch(rep) {
             case 0:
-                addr = dynarec64_0F(dyn, addr, ip, ninst, rex, ok, need_epilog);
+                if (rv64_vector) {
+                    retaddr = dynarec64_0F_vector(dyn, addr, ip, ninst, rex, ok, need_epilog);
+                    addr = retaddr ? retaddr : dynarec64_0F(dyn, addr, ip, ninst, rex, ok, need_epilog);
+                } else
+                    addr = dynarec64_0F(dyn, addr, ip, ninst, rex, ok, need_epilog);
                 break;
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_00_0.c:186]: You can probably simplify with
```C
                if (rv64_vector) 
                    retaddr = dynarec64_0F_vector(dyn, addr, ip, ninst, rex, ok, need_epilog);
                addr = retaddr ? retaddr : dynarec64_0F(dyn, addr, ip, ninst, rex, ok, need_epilog);
```

File: src/dynarec/rv64/dynarec_rv64_660f_vector.c (hunk 1 above)
Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_660f_vector.c:153]: no need, vtype is reset by the following `SET_ELEMENT_WIDTH`.

File: src/dynarec/rv64/dynarec_rv64_00_0.c (hunk 2 above)
Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_00_0.c:186]: indeed.

**Review by ptitSeb — APPROVED** [review]

**Pull request merged**